src/tess_sip/tess_sip.py

```diff
@@ -92,65 +99,101 @@ def SIP(
+            tpfs_uncorr = [
+                (tpf + np.nan_to_num(tpf.flux_bkg.value))[
+                    np.isfinite(np.nansum(tpf.flux_bkg.value, axis=(1, 2)))
+                ]
+                for tpf in tpfs
             ]
-            for tpf in tpfs
-        ]
-    else:
-        tpfs_uncorr = tpfs
+        else:
+            tpfs_uncorr = tpfs
 
-    apers = [
-        tpf.pipeline_mask
-        if tpf.pipeline_mask.any()
-        else tpf.create_threshold_mask(aperture_threshold)
-        for tpf in tpfs_uncorr
-    ]
-    bkg_apers = [
-        (~aper) & (np.nansum(tpf.flux, axis=0) != 0)
-        for aper, tpf in zip(apers, tpfs_uncorr)
-    ]
-    lc = (
-        lk.LightCurveCollection(
-            [
-                tpf.to_lightcurve(aperture_mask=aper)
-                for tpf, aper in zip(tpfs_uncorr, apers)
-            ]
+        apers = [
+            tpf.pipeline_mask
+            if tpf.pipeline_mask.any()
+            else tpf.create_threshold_mask(aperture_threshold)
+            for tpf in tpfs_uncorr
+        ]
+        bkg_apers = [
+            (~aper) & (np.nansum(tpf.flux, axis=0) != 0)
+            for aper, tpf in zip(apers, tpfs_uncorr)
+        ]
+        lc = (
+            lk.LightCurveCollection(
+                [
+                    tpf.to_lightcurve(aperture_mask=aper)
+                    for tpf, aper in zip(tpfs_uncorr, apers)
+                ]
+            )
+            .stitch(lambda x: x)
+            .normalize()
         )
-        .stitch(lambda x: x)
-        .normalize()
-    )
-    lc.flux_err.value[~np.isfinite(lc.flux_err.value)] = np.nanmedian(lc.flux_err.value)
+        lc.flux_err.value[~np.isfinite(lc.flux_err.value)] = np.nanmedian(lc.flux_err.value)
 
-    # Run the same routines on the background pixels
-    lc_bkg = (
-        lk.LightCurveCollection(
-            [
-                tpf.to_lightcurve(aperture_mask=bkg_aper)
-                for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)
-            ]
+        # Run the same routines on the background pixels
+        lc_bkg = (
+            lk.LightCurveCollection(
+                [
+                    tpf.to_lightcurve(aperture_mask=bkg_aper)
+                    for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)
+                ]
+            )
+            .stitch(lambda x: x)
+            .normalize()
         )
-        .stitch(lambda x: x)
-        .normalize()
-    )
-    lc_bkg.flux_err.value[~np.isfinite(lc_bkg.flux_err.value)] = np.nanmedian(
-        lc_bkg.flux_err.value
-    )
+        lc_bkg.flux_err.value[~np.isfinite(lc_bkg.flux_err.value)] = np.nanmedian(
+            lc_bkg.flux_err.value
+        )
+        
+    # Check if tpfs is a lightcurve file rather than a target pixel file
+    elif ((type(tpfs) is lk.lightcurve.TessLightCurve) or 
+            (type(tpfs) is lk.collections.LightCurveCollection)):
+        print("Lightcurve File Input")
+        for lcf in tpfs:
```

Comment:
I feel that this may break if you pass in a `LightCurve` instance. 

If this is a `LightCurveCollection` then this will iterate through each light curve. If this is a `LightCurve`, this will iterate through each element in the light curve. Make sure there is an error catch above this to force the input data to be some sort of `Collection`.